tasks/prelim.yml

```diff
@@ -1,110 +1,249 @@
+  changed_when: false
+  failed_when: false
+  check_mode: false
+  register: amazon2cis_sudoers_files
+
+- name: "PRELIM | Section 5.2.x | Configure System Accounting (auditd)"
+  ansible.builtin.package:
       name: audit
       state: present
   vars:
       ansible_python_interpreter: /bin/python
 
-- name: "PRELIM | Section 5.1 | Configure cron"
-  package:
+- name: "PRELIM | 5.2.4.x | Ensure audit log files are mode 0640 or less permissive | discover file"
+  ansible.builtin.shell: "grep ^log_file /etc/audit/auditd.conf | awk '{ print $NF }'"
+  changed_when: false
+  failed_when: audit_discovered_logfile.rc not in [0, 1]
+  register: audit_discovered_logfile
+  when:
+      - amazon2cis_rule_5_2_4_1
+      - amazon2cis_rule_5_2_4_2
+      - amazon2cis_rule_5_2_4_3
+      - amazon2cis_rule_5_2_4_4
+  tags:
+      - level2-server
+      - level2-workstation
+      - patch
+      - auditd
+      - rule_5.2.4.1
+      - rule_5.2.4.2
+      - rule_5.2.4.3
+      - rule_5.2.4.4
+
+- name: "PRELIM | 5.2.4.5/6/7 | Audit conf and rules files | list files"
+  ansible.builtin.find:
+      path: /etc/audit
+      file_type: file
+      recurse: true
+      patterns: '*.conf,*.rules'
+  register: auditd_conf_files
+  when:
+      - amazon2cis_rule_5_2_4_5 or
+        amazon2cis_rule_5_2_4_6 or
+        amazon2cis_rule_5_2_4_7
+  tags:
+      - level2-server
+      - level2-workstation
+      - patch
+      - auditd
+      - rule_5.2.4.5
+      - rule_5.2.4.6
+      - rule_5.2.4.7
+
+- name: "PRELIM | Section 4.1.1.x | Configure cron"
+  ansible.builtin.package:
       name: cronie
       state: present
   vars:
       ansible_python_interpreter: /bin/python
 
-### NOTE: You will need to adjust the UID range in parenthases below.
-### ALSO NOTE: We weed out any user with a home dir not in standard locations because interactive users shouldn't have those paths as a home dir. Add or removed directory paths as needed below.
-- name: "PRELIM | 6.2.8 | Gather local interactive user directories"
-  shell: "getent passwd {1000..65535} | cut -d: -f6 | sort -u | grep -v '/var/' | grep -v '/nonexistent/*' | grep -v '/run/*'"
+- name: "PRELIM | Interactive User accounts home directories"
+  tags:
+      - always
+  ansible.builtin.shell: >
+      grep -E -v '^(root|halt|sync|shutdown)' /etc/passwd | awk -F: '(!index($7, "sbin/nologin") && $7 != "/bin/false") { print $6 }'
   changed_when: false
-  failed_when: false
-  register: amazon2cis_6_2_8_getent
+  register: interactive_users_home
+
+##### Optional #####
+
+- name: "PRELIM | Optional | If IPv6 disable to stop ssh listening"
+  when:
+      - amazon2cis_ipv6_sshd_disable
+      - not amazon2cis_ipv6_required
   tags:
-      - skip_ansible_lint
-      - rule_6.2.8
+      - always
+  ansible.builtin.lineinfile:
+      path: /etc/ssh/sshd_config
+      regexp: ^AddressFamily
+      line: AddressFamily inet
+  notify: Restart_sshd
```

Comment:
this handler doesn't exist - it seems to be called `restart sshd`

Reply:
Thanks - appreciate it. Yeah I realise but there was less friction in adding this comment than filling out the new issue template ;)